### PR TITLE
Update quickstart to offer cloud and local options

### PR DESCRIPTION
**Pull request by jseldess**
Fixes DOC-3004.

This PR adds instructions for local demo clusters to the revised Quickstart. It also adds instructions on how to connect using the SQL CLI for Serverless, Dedicated, and Self Hosted clusters as that info is no longer in the Quickstart.

### CODE REVIEW

**Review by ericharmeling — COMMENTED** [review]
Looks good! I did have a few comments.

> Reviewed 1 of 6 files at r1, 1 of 5 files at r3, 10 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12808#-:-MzG2RxI0DZf0Z08FLQj:b-3vxvds)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans and @jseldess)

___
*[_includes/cockroachcloud/quickstart/database-url-environment-variable.md, line 1 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFrRg9814WJl_VIsVW:-MzFrRg9814WJl_VIsVX:bjbyod4) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/_includes/cockroachcloud/quickstart/database-url-environment-variable.md#L1)):*
> ```Markdown
> 1. Set the `DATABASE_URL` environment variable to the connection string that you copied in Step 2:
> ```

Nit: avoid hardcoding "Step 2" 



_Suggestion:_
```Markdown
 earlier:
```
___
*[cockroachcloud/learn-cockroachdb-sql.md, line 14 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFrxVKBvVwFyHKsSMk:-MzFrxVKBvVwFyHKsSMl:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/cockroachcloud/learn-cockroachdb-sql.md#L14)):*
> ```Markdown
> ## Before you begin
> 
> Make sure you have already [create a free {{ site.data.products.db }} cluster](quickstart.html) and [connected the SQL client](../{{site.versions["stable"]}}/connect-to-the-database.html?filters=sql).
> ```





_Suggestion:_
```Markdown
created a free
```
___
*[cockroachcloud/learn-cockroachdb-sql.md, line 14 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFs06B3AlT1xBnrtaq:-MzFs06B3AlT1xBnrtar:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/cockroachcloud/learn-cockroachdb-sql.md#L14)):*
> ```Markdown
> ## Before you begin
> 
> Make sure you have already [create a free {{ site.data.products.db }} cluster](quickstart.html) and [connected the SQL client](../{{site.versions["stable"]}}/connect-to-the-database.html?filters=sql).
> ```





_Suggestion:_
```Markdown
connected to the
```
___
*[cockroachcloud/quickstart.md, line 142 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFsk_I2PIFijUWuIyX:-MzFsk_J3_6QcoDG-A57:b57fbd4) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/cockroachcloud/quickstart.md#L142)):*
> ```Markdown
> <section class="filter-content" markdown="1" data-scope="java">
> 
> 1. Use the `cockroach convert-url` command to convert the connection string that you copied in Step 2 to a [valid connection string for JDBC connections](../stable/connect-to-the-database.html?filters=java):
> ```

Nit: avoid hardcoding "Step 2"



_Suggestion:_
```Markdown
 earlier
```
___
*[v21.2/connect-to-the-database.md, line 38 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFuyU9-LlRu_uk33sl:-MzFuyU9-LlRu_uk33sm:b-riogyn) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L38)):*
> ```Markdown
>   <button class="filter-button page-level" data-scope="java">Java</button>
>   <button class="filter-button page-level" data-scope="ruby">Ruby</button>
>   <button class="filter-button page-level" data-scope="sql">SQL client</button>
> ```

TBH, I'm on the fence about adding a "SQL client" section to this page, for a couple reasons:

1. The `cockroach` CLI is kind of out of place on this page. It isn't a third-party driver for connecting programmatically to CRDB, like the rest of these.
2. I don't personally think we need any more documentation for connecting with the CLI. There are already multiple reference pages on connecting with `cockroach` commands (e.g., https://deploy-preview-12808--cockroachdb-docs.netlify.app/docs/v21.2/connection-parameters and https://deploy-preview-12808--cockroachdb-docs.netlify.app/docs/v21.2/cockroach-sql.html), and a lot of other pages demonstrate connecting with the CLI. Also, the cloud console gives instructions for connecting with the CLI, and the CLI help text is very descriptive.

That might seem nit-picky though. Your call.
___
*[v21.2/connect-to-the-database.md, line 1239 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzG-fLi9D5dM2JzU5xO:-MzG-fLi9D5dM2JzU5xP:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L1239)):*
> ```Markdown
> <div class="filter-content" markdown="1" data-scope="sql">
> 
> <div class="filter-content" markdown="1" data-scope="serverless">
> ```





_Suggestion:_
```Markdown
<div class="filter-content" markdown="1" data-scope="serverless">

To connect to CockroachDB with the [CockroachDB SQL shell](cockroach-sql.html):
```
___
*[v21.2/connect-to-the-database.md, line 1248 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFxw9n8JAHF81RQrCI:-MzFxw9n8JAHF81RQrCJ:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L1248)):*
> ```Markdown
> 
> 1. Choose your OS from the drop-down.
> 1. Follow the steps in the **Command Line** tab to connect to your cluster using [`cockroach sql`](cockroach-sql.html).
> ```





_Suggestion:_
```Markdown
 to download the `cockroach` binary and connect to your cluster using the [`cockroach sql`](cockroach-sql.html) CLI command.
```
___
*[v21.2/connect-to-the-database.md, line 1252 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzG-lgSCPTESfYt7k2w:-MzG-lgSCPTESfYt7k2x:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L1252)):*
> ```Markdown
> </div>
> 
> <div class="filter-content" markdown="1" data-scope="dedicated">
> ```





_Suggestion:_
```Markdown
<div class="filter-content" markdown="1" data-scope="dedicated">

To connect to CockroachDB with the [CockroachDB SQL shell](cockroach-sql.html):
```
___
*[v21.2/connect-to-the-database.md, line 1254 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzFy6mG1DFxP37c0MJo:-MzFy6mG1DFxP37c0MJp:bx11f7j) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L1254)):*
> ```Markdown
> <div class="filter-content" markdown="1" data-scope="dedicated">
> 
> Follow steps 1-5 in the [{{ site.data.products.dedicated }} quickstart.](../cockroachcloud/quickstart-trial-cluster.html) to connect to your cluster using [`cockroach sql`](cockroach-sql.html).
> ```

Shouldn't we just list the minimum steps to connect to dedicated, like we do for serverless above?



_Code quote:_
```Markdown
Follow steps 1-5 in the [{{ site.data.products.dedicated }} quickstart.](../cockroachcloud/quickstart-trial-cluster.html) to connect to your cluster using [`cockroach sql`](cockroach-sql.html).
```
___
*[v21.2/connect-to-the-database.md, line 1260 at r4](https://reviewable.io/reviews/cockroachdb/docs/12808#-MzG-pTC2z2zfpR__2zz:-MzG-pTC2z2zfpR__3--:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/620f2743976efdf2c20ddf37d66587cdb6bf9ce4/v21.2/connect-to-the-database.md#L1260)):*
> ```Markdown
> <div class="filter-content" markdown="1" data-scope="core">
> 
> To connect to a node running on your local machine open a terminal and enter:
> ```





_Suggestion:_
```Markdown
To connect to a CockroachDB node running on your local machine with the [CockroachDB SQL shell](cockroach-sql.html), open a terminal and enter:
```

File: cockroachcloud/quickstart.md
Comment (ericharmeling) [inline cockroachcloud/quickstart.md:37]: Is there a reason why this isn't a step? It looks a little odd now, having this "task" as a floating level 3 header.

**Review by ericharmeling — APPROVED** [review]
LGTM modulo one nit about a header

File: cockroachcloud/quickstart.md
Comment (ianjevans) [reply cockroachcloud/quickstart.md:37]: Yes. Since we hard-code the step numbers in the headers, the number of steps for Serverless was one more than for demo, so the shared headers wouldn't match.

I could pull every common step's content out into an include, but that makes it harder to edit.

**Review by emily-horing — COMMENTED** [review]
Is removing dedicated from the quickstart and putting it in a separate quickstart new? I'm not sure if that's ideal, maybe @kannanlakshmi has an opinion on that?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12808#-:-N0WPnwg7BlWFibI-lyn:b-vmr1oz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @ianjevans, and @jseldess)

**Review by jseldess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12808)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @ianjevans, and @jseldess)

___
*[`cockroachcloud/quickstart.md` line 37 at r5](https://reviewable.io/reviews/cockroachdb/docs/12808#-MtyeHHATZRrYxULdUt4-r5-37:-N0WaI4iDz5PfXD4Gf73:b-h6u5al) ([raw file](https://github.com/cockroachdb/docs/blob/aa68129f21f670d21f1dd3ff13210b4137d62a46/cockroachcloud/quickstart.md#L37)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

Makes sense! Thanks for the reply.
</blockquote></details>

Since the Create SQL User dialog displays automatically, based on the last substep of Step 1, should we consider just merging this all into "Step 1. Create a free cluster"? Or will that not work with other places the include is being used?

File: cockroachcloud/quickstart.md
Comment (jseldess) [inline cockroachcloud/quickstart.md:85]: Instead of having 1 substep here, can we just make this substep 1 of "Start CockroachDB"?

File: cockroachcloud/quickstart.md
Comment (jseldess) [inline cockroachcloud/quickstart.md:9]: The liquid variable is translating to "CockroachDB Cloud", which isn't correct for when you select to work with a local cluster. Can we just use "CockroachDB"?

**Comment by jseldess** [comment]
@emily-horing, @kannanlakshmi, in terms of removing dedicated from the quickstart, the idea is to lead with the options that are quickest and most likely to engage devs just getting started with crdb. Thus, cloud (Serverless) and local (demo). 

That said, we really should find a way to make it more explicit that you're using Serverless, and also find a way to point people to the Dedicated how-to. @ianjevans, do you think we can take a swing at this in the note at the top of the "Use a cloud cluster" tab?

File: cockroachcloud/quickstart.md
Comment (ianjevans) [reply cockroachcloud/quickstart.md]: Done.

File: cockroachcloud/quickstart.md
Comment (ianjevans) [reply cockroachcloud/quickstart.md]: Since Jesse suggested consolidating the demo steps, I've conditionalized all the step headings, so I reverted this back to its own step.

File: cockroachcloud/quickstart.md
Comment (ianjevans) [reply cockroachcloud/quickstart.md]: > > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12808)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @ianjevans, and @jseldess)
> 
> _[`cockroachcloud/quickstart.md` line 37 at r5](https://reviewable.io/reviews/cockroachdb/docs/12808#-MtyeHHATZRrYxULdUt4-r5-37:-N0WaI4iDz5PfXD4Gf73:b-h6u5al) ([raw file](https://github.com/cockroachdb/docs/blob/aa68129f21f670d21f1dd3ff13210b4137d62a46/cockroachcloud/quickstart.md#L37)):_
> 
> Previously, ericharmeling (Eric Harmeling) wrote…
> Since the Create SQL User dialog displays automatically, based on the last substep of Step 1, should we consider just merging this all into "Step 1. Create a free cluster"? Or will that not work with other places the include is being used?

I think I'd rather keep the include as is, so I've conditionalized all the step headings and turned this back into a separate step section.

> The liquid variable is translating to "CockroachDB Cloud", which isn't correct for when you select to work with a local cluster. Can we just use "CockroachDB"?

Missed this before. Done.

**Comment by ianjevans** [comment]
> @emily-horing, @kannanlakshmi, in terms of removing dedicated from the quickstart, the idea is to lead with the options that are quickest and most likely to engage devs just getting started with crdb. Thus, cloud (Serverless) and local (demo).
> 
> That said, we really should find a way to make it more explicit that you're using Serverless, and also find a way to point people to the Dedicated how-to. @ianjevans, do you think we can take a swing at this in the note at the top of the "Use a cloud cluster" tab?

Done.

File: _includes/cockroachcloud/quickstart/database-url-environment-variable.md
Comment (jseldess) [inline _includes/cockroachcloud/quickstart/database-url-environment-variable.md:2]: Given this is just one step, should we get rid of the numbering?

File: _includes/cockroachcloud/quickstart/database-url-environment-variable.md
Comment (jseldess) [inline _includes/cockroachcloud/quickstart/database-url-environment-variable.md:14]: Same as above.

**Review by jseldess — COMMENTED** [review]
LGTM, with one more suggestion.

File: _includes/cockroachcloud/quickstart/database-url-environment-variable.md
Comment (ianjevans) [reply _includes/cockroachcloud/quickstart/database-url-environment-variable.md]: Done.

**Pull request merged**
